### PR TITLE
CUS-33043/amazon-widget-amp

### PR DESCRIPTION
**Pull request by ricardoribasmrf**

### CODE REVIEW

File: json-lint/schemas/inventory/adservers/a9.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      "description": "contains required keys",
+      "body": {
+        "type": "a9",
+        "data-amzn_assoc_ad_mode": "$1"
+      }
+    }
+  ],
+  "properties": {
+    "type": {
+      "enum": [
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/a9.json]: const //

File: json-lint/schemas/inventory/adservers/a9.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{
+  "$id": "#a9",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "title": "Amazon adserver",
+  "description": "https://github.com/ampproject/amphtml/blob/master/ads/a9.md",
+  "additionalProperties": false,
+  "required": [
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/a9.json]: data-adinstanceid ??  
I guess this is totally required, if the goal is to make money 😆  🔢

File: json-lint/schemas/inventory/adservers/a9.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    },
+    "data-adinstanceid": {
+      "type": "string"
+    },
+    "data-regionurl": {
+      "type": "string"
+    },
+    "data-divid": {
+      "type": "string"
+    }
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/a9.json]: can we add the remaining keys that are valid for the amp-ad ? 
https://github.com/ampproject/amphtml/blob/master/ads/a9.md

data-amzn_assoc_asins
data-amzn_assoc_placement
data-amzn_assoc_search_bar     >> boolean
data-amzn_assoc_tracking_id
data-amzn_assoc_search_bar_position
data-amzn_assoc_ad_mode
data-amzn_assoc_ad_type
data-amzn_assoc_marketplace
data-amzn_assoc_region
data-amzn_assoc_title
data-amzn_assoc_default_search_phrase
data-amzn_assoc_default_category
data-amzn_assoc_linkid
data-regionurl

File: json-lint/schemas/inventory/adservers/zergnet.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      "description": "contains required keys",
+      "body": {
+        "type": "zergnet",
+        "data-zergid": "$1"
+      }
+    }
+  ],
+  "properties": {
+    "type": {
+      "enum": [
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/zergnet.json]: const

File: json-lint/schemas/inventory/adservers/a9.json (hunk 2 above)
Comment (ricardoribasmrf) [reply json-lint/schemas/inventory/adservers/a9.json]: I can put it mandatory, but the documentation only states that one

Comment (ricardoribasmrf) [reply json-lint/schemas/inventory/adservers/a9.json]: In some examples, we don't have instanceId. However, as talked offline, if we need to remove it, should be quick.

**Comment by ricardoribasmrf** [comment]
❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️

**Pull request merged**